test/src/main/java/org/springframework/security/test/web/servlet/response/SecurityMockMvcResultHandlers.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.test.web.servlet.response;
+
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.test.context.TestSecurityContextHolder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultHandler;
+
+/**
+ * Security related {@link MockMvc} {@link ResultHandler}s
+ *
+ * @author Marcus da Coregio
+ * @since 5.6
+ */
+public final class SecurityMockMvcResultHandlers {
+
+	private SecurityMockMvcResultHandlers() {
+	}
+
+	/**
+	 * Exports the {@link SecurityContext} from {@link TestSecurityContextHolder} to
+	 * {@link SecurityContextHolder}
+	 */
+	public static ResultHandler exportSecurityContext() {
```

Comment:
I'd like a name that is a bit more explicit as to what is happening. Perhaps `propagateTestSecurityContext`? I think a better name could be found that my suggestion, but the idea is it should convey what is being done.

Reply:
"propagate", "retain" and "keep" all seem to imply that it will be retained across multiple `MockMvc` invocations in the same test. Would that be the case? If this isn't a concern, then my favorite among these options is "retain". If it is a concern, then I still like "export" best.

Looking more deeply into the support, I see that there is a request post-processor static method called `testSecurityContext()`. In that case, "xyzTestSecurityContext" does seem more consistent.

Reply:
I think we should indeed mention `TestSecurityContext` in the name. It would be explicit if we do `exportTestSecurityContext`, since it gives the idea of the `TestSecurityContext` being exported somewhere, and this also gives the opportunity to extend it to `exportTestSecurityContext(Consumer<SecurityContext> exportTo)`.

If we use names like **save**, **propagate**, **preserve**, all of them give the idea that the `TestSecurityContextHolder` will not be cleared out.

Reply:
> "propagate", "retain" and "keep" all seem to imply that it will be retained across multiple MockMvc invocations in the same test. Would that be the case? 

I am not a fan of save, retain, preserve, or keep because I interpret these to mean that the `TestSecurityContextHolder` will not be changed which is true regardless of the `ResultHandler` being used. The behavior is copy/export/propagate the `TestSecurityContextHolder` into the `SecurityContext`.